### DISCUSSION TITLE
How to name composables in subfolders to make auto-import work?

### DISCUSSION THREAD

**Discussion by alex00888** [nuxt/nuxt#18253] [Questions]
I need to group my composables in subfolders. But auto-import is not working.
`composables/testComposable.ts` imports fine, but `composables/test/composable.ts` is not auto-imported. I tried `composables/test/Composable.ts` but it's not working too :(

**Comment by wobsoriano** [comment]
You can create an `index.ts` in your composables folder then reexport files inside `test` folder:

```ts
// composables/test/useUserTest.ts
import { useState } from '#app'

export const useUserTest = () => {
  return useState('test', () => ({
      name: 'test',
      email: 'test@yahoo.com'
  }))
}
```

```ts
// composables/index.ts
export { useUserTest } from './test/useUserTest'
```

```html
<script setup lang="ts">
  const test = useUserTest()

  console.log(test.value)
</script>
```
